tests/test_zinolib_controllers_zino1.py

```diff
@@ -144,3 +145,102 @@ def test_get_log_for_id(self):
+        zino1 = FakeZino1EventManager.configure(None)
+        return zino1
+
+    def test_cmd_scavenged(self):
+        zino1 = self.init_manager()
+        zino1.get_events()
+        self.assertIn(raw_event_id, zino1.events)
+        self.assertNotIn(raw_event_id, zino1.removed_ids)
+        updates = UpdateHandler(zino1)
+        update = NotifierResponse(raw_event_id, "","")
+        ok = updates.cmd_scavenged(update)
+        self.assertTrue(ok)
+        self.assertNotIn(raw_event_id, zino1.events)
+        self.assertIn(raw_event_id, zino1.removed_ids)
+
+    def test_cmd_attr(self):
+        zino1 = self.init_manager()
+        zino1.get_events()
+        old_events = zino1.events.copy()
+        old_events[raw_event_id].priority = 500
+        updates = UpdateHandler(zino1)
+        update = NotifierResponse(raw_event_id, "","")
+        ok = updates.cmd_attr(update)
+        self.assertTrue(ok)
+        self.assertNotEqual(zino1.events[raw_event_id].priority, old_events[raw_event_id].priority)
+
+    def test_cmd_state_is_closed_and_autoremove_is_on(self):
+        zino1 = self.init_manager()
+        zino1.get_events()
+        self.assertNotIn(raw_event_id, zino1.removed_ids)
+        self.assertIn(raw_event_id, zino1.events)
+        updates = UpdateHandler(zino1, autoremove=True)
+        update = NotifierResponse(raw_event_id, "", "X closed")
+        ok = updates.cmd_state(update)
+        self.assertTrue(ok)
+        self.assertIn(raw_event_id, zino1.removed_ids)
+        self.assertNotIn(raw_event_id, zino1.events)
+
+    def test_cmd_state_is_closed_and_autoremove_is_off(self):
+        zino1 = self.init_manager()
+        zino1.get_events()
+        old_events = zino1.events.copy()
+        old_events[raw_event_id].priority = 500
+        updates = UpdateHandler(zino1, autoremove=False)
+        update = NotifierResponse(raw_event_id, "","X closed")
+        ok = updates.cmd_state(update)
+        self.assertTrue(ok)
+        self.assertNotEqual(zino1.events[raw_event_id].priority, old_events[raw_event_id].priority)
+
+    def test_cmd_state_is_not_closed(self):
+        zino1 = self.init_manager()
+        zino1.get_events()
+        old_events = zino1.events.copy()
+        old_events[raw_event_id].priority = 500
+        updates = UpdateHandler(zino1, autoremove=False)
+        update = NotifierResponse(raw_event_id, "","x butterfly")
+        ok = updates.cmd_state(update)
+        self.assertTrue(ok)
+        self.assertNotEqual(zino1.events[raw_event_id].priority, old_events[raw_event_id].priority)
+
+    def test_fallback(self):
+        zino1 = self.init_manager()
+        updates = UpdateHandler(zino1)
+        update = NotifierResponse(raw_event_id, "", "")
+        with self.assertLogs('zinolib.controllers.zino1', level='WARNING') as cm:
+            self.assertFalse(updates.fallback(update))
+            self.assertEqual(cm.output, ['WARNING:zinolib.controllers.zino1:Unknown update type: "" for id 139110'])
```

Comment:
I feel like we should be very careful about testing exact log output.  Is it a functional requirement that fallback produces a log message formatted *exactly* like this?